ql/instruments/swaption.hpp

```diff
@@ -97,10 +102,13 @@ namespace QuantLib {
         Settlement::Method settlementMethod() const {
             return settlementMethod_;
         }
-        Swap::Type type() const { return swap_->type(); }
+        Swap::Type type() const { return swap_ ? swap_->type() : swapOis_->type(); }
```

Comment:
If `swapOis_` is assumed to be non-null when `swap_` is null then this should probably be asserted in the constructor.

Reply:
It is implicitly enforced by the signature. One can never pass both a vanillaswap and a overnightindexedswap to the constructor.